diff --git a/funcr/funcr.go b/funcr/funcr.go
index 8ecf434..9ee5caa 100644
--- a/funcr/funcr.go
+++ b/funcr/funcr.go
@@ -722,6 +722,7 @@ func (f *Formatter) AddName(name string) {
 func (f *Formatter) AddValues(kvList []interface{}) {
 	// Three slice args forces a copy.
 	n := len(f.values)
+	// Preserve raw saved values before rendering them for output.
 	vals := f.values[:n:n]
 	vals = append(vals, kvList...)
 	if hook := f.opts.RenderValuesHook; hook != nil {
