diff --git a/randomization.go b/randomization.go
index 95d91b2..7fac027 100644
--- a/randomization.go
+++ b/randomization.go
@@ -9,6 +9,9 @@ var randomFloat64 = rand.Float64
 
 // randomizedInterval returns the next delay using the configured jitter.
 func randomizedInterval(randomizationFactor float64, currentInterval time.Duration) time.Duration {
+	if randomizationFactor == 0 {
+		return currentInterval
+	}
 	return getRandomValueFromInterval(
 		randomizationFactor,
 		randomFloat64(),
