diff --git a/src/itsdangerous/timed.py b/src/itsdangerous/timed.py
--- a/src/itsdangerous/timed.py
+++ b/src/itsdangerous/timed.py
@@ -151,7 +151,7 @@ class TimestampSigner(Signer):
         if ts_int is None:
             raise BadTimeSignature("Malformed timestamp", payload=value)
 
         # Check timestamp is not older than max_age.
-        if max_age:
+        if max_age is not None:
             age = self.get_timestamp() - ts_int
             self._validate_timestamp_age(age, max_age, value, ts_int)
 
