diff --git a/src/itsdangerous/timed.py b/src/itsdangerous/timed.py
--- a/src/itsdangerous/timed.py
+++ b/src/itsdangerous/timed.py
@@ -50,6 +50,23 @@ class TimestampSigner(Signer):
         value = value + sep + timestamp
         return value + sep + self.get_signature(value)
 
+    def _validate_timestamp_age(
+        self, age: int, max_age: int, value: bytes, timestamp: int
+    ) -> None:
+        if age > max_age:
+            raise SignatureExpired(
+                f"Signature age {age} > {max_age} seconds",
+                payload=value,
+                date_signed=self.timestamp_to_datetime(timestamp),
+            )
+
+        if age < 0:
+            raise SignatureExpired(
+                f"Signature age {age} < 0 seconds",
+                payload=value,
+                date_signed=self.timestamp_to_datetime(timestamp),
+            )
+
     # Ignore overlapping signatures check, return_timestamp is the only
     # parameter that affects the return type.
 
@@ -134,23 +151,10 @@ class TimestampSigner(Signer):
         if ts_int is None:
             raise BadTimeSignature("Malformed timestamp", payload=value)
 
-        # Check timestamp is not older than max_age
-        if max_age is not None:
+        # Check timestamp is not older than max_age.
+        if max_age:
             age = self.get_timestamp() - ts_int
-
-            if age > max_age:
-                raise SignatureExpired(
-                    f"Signature age {age} > {max_age} seconds",
-                    payload=value,
-                    date_signed=self.timestamp_to_datetime(ts_int),
-                )
-
-            if age < 0:
-                raise SignatureExpired(
-                    f"Signature age {age} < 0 seconds",
-                    payload=value,
-                    date_signed=self.timestamp_to_datetime(ts_int),
-                )
+            self._validate_timestamp_age(age, max_age, value, ts_int)
 
         if return_timestamp:
             return value, self.timestamp_to_datetime(ts_int)
