diff --git a/src/click/utils.py b/src/click/utils.py
index d680f01..a8021c9 100644
--- a/src/click/utils.py
+++ b/src/click/utils.py
@@ -49,7 +49,8 @@ def _normalize_token(
 def _normalize_opt(
     opt: str, normalize: t.Callable[[str], str] | None
 ) -> str:
-    return _normalize_token(opt, normalize)
+    prefix, value = _split_opt(opt)
+    return f"{prefix}{_normalize_token(value, normalize)}"
 
 
 def _posixify(name: str) -> str:
